pkg/ansible/controller/reconcile.go

```diff
@@ -310,7 +310,7 @@ func (r *AnsibleOperatorReconciler) markDone(u *unstructured.Unstructured, names
 	// Get the latest resource to prevent updating a stale status
 	err := r.Client.Get(context.TODO(), namespacedName, u)
 	if apierrors.IsNotFound(err) {
-		logger.Info("Resource not found, assuming it was deleted", err)
+		logger.Info("Resource not found, assuming it was deleted")
```

Comment:
Hi @joelanford, 

It shows not be the root cause of this issue because the Info should not require a par, just the error one. The info expected only a string. 

Also, note that when the CR is deleted we should stop the reconcile which shows that is not what is happening in these scenarios. The markDone and markError will return nil then the reconcile will not have the info to required to be stoped. 

IMHO: In these places, if an error is faced then it should be returned. Where it is called should use the `IsNotFound(err)` in order to check if is the scenario where the CR was deleted to stop the reconcile. 

Also, I would add some comments as follows to make it clear. 

	if err != nil {
		if errors.IsNotFound(err) {
			// Request object not found, could have been deleted after reconcile request.
			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
			// Return and don't requeue
			logger.Info("Resource not found, assuming it was deleted")
			return reconcile.Result{}, nil
		}
		// Error reading the object - requeue the request.
		logger.Error(err, "Failed to get resource.")
		return reconcile.Result{}, err
	}

Reply:
@camilamacedo86 I made this PR to solve just the DPANIC issue from #2025, and I specifically didn't want to include any other ancillary changes.

Based on my quick look at #2032, it looks like just a slight refactoring with some extra logging, which isn't related to the issue reported in #2025 as far as I can tell.

Reply:
The  #2032 would allow the reconcile to stop when the CR is removed which by looking the code in the markDone and markError is not.